dumpling/export/sql.go

```diff
@@ -57,59 +59,63 @@ func ShowTables(db *sql.Conn) ([]string, error) {
 
 // ShowCreateDatabase constructs the create database SQL for a specified database
 // returns (createDatabaseSQL, error)
-func ShowCreateDatabase(db *sql.Conn, database string) (string, error) {
+func ShowCreateDatabase(tctx *tcontext.Context, db *BaseConn, database string) (string, error) {
 	var oneRow [2]string
 	handleOneRow := func(rows *sql.Rows) error {
 		return rows.Scan(&oneRow[0], &oneRow[1])
 	}
 	query := fmt.Sprintf("SHOW CREATE DATABASE `%s`", escapeString(database))
-	err := simpleQuery(db, query, handleOneRow)
-	if mysqlErr, ok := errors.Cause(err).(*mysql.MySQLError); ok {
-		// Falling back to simple create statement for MemSQL/SingleStore, because of this:
-		// ERROR 1706 (HY000): Feature 'SHOW CREATE DATABASE' is not supported by MemSQL.
-		if mysqlErr.Number == 1706 {
-			return fmt.Sprintf("CREATE DATABASE `%s`", escapeString(database)), nil
+	err := db.QuerySQL(tctx, handleOneRow, func() {
+		oneRow[0], oneRow[1] = "", ""
+	}, query)
+	if multiErrs := multierr.Errors(err); len(multiErrs) > 0 {
+		for _, multiErr := range multiErrs {
+			if mysqlErr, ok := errors.Cause(multiErr).(*mysql.MySQLError); ok {
+				// Falling back to simple create statement for MemSQL/SingleStore, because of this:
+				// ERROR 1706 (HY000): Feature 'SHOW CREATE DATABASE' is not supported by MemSQL.
```

Comment:
this error code is not stable among different MySQL-like databases. Maybe we can use error message to match some error

https://mariadb.com/kb/en/mariadb-error-codes/

Reply:
It's mainly used for `MemSQL/SingleStore`. I have changed the match string to `SHOW CREATE DATABASE`.